### ISSUE TITLE
Double confirm alert

### ISSUE THREAD

**Issue by DarkSide666** [atk4/ui#649]
```
$this->add(['Button', 'Disconnect', 'icon'=>'remove'])
    ->on('click', function(){
        ...
    }, ['confirm'=>'Do you really want to disconnect?'])
;
```
This shows confirm dialog 2 times. Shows it and then when you click OK button shows it one more time.

**Comment by abbadon1334** [comment]
i'm on it.
the problem is in generated javascript code

```
$("#_a1183b66__page_crud_button").on("click",function(event) {
--
event.preventDefault();
event.stopPropagation();
if(!confirm("Do you really want to disconnect?")){return;};
$(this).atkAjaxec({"uri":"6?atk_backend_main_course_tabs_tabssubview_2_virtualpage=cut\x26_a1183b66__page_crud_button_click=ajax\x26__atk_callback=1","uri_options":[],"confirm":"Do you really want to disconnect?"});
});

```
this is the first confirm 
```
if(!confirm("Do you really want to disconnect?")){return;};
```
and the second come from this
```
"confirm":"Do you really want to disconnect?"});
```
i debug the function **View->on** with xdebug and $defaults are correctly set

i try to comment #642 changes in ***View->on*** and the generated code is correct

from what i see jsCallback set a confirm indipendent by the logic of View->on.

i look in the history and found that #642 add this bug, adding a confirm prompt to any callback.

to maintain this feature on all callback and use it like #642 i try to comment **confirm** in **atkAjaxec** defaults
``
    public function jsRender()
    {
        if (!$this->app) {
            throw new Exception(['Call-back must be part of a RenderTree']);
        }

        return (new jQuery())->atkAjaxec([
            'uri'         => $this->getJSURL(),
            'uri_options' => $this->args,
            //'confirm'     => $this->confirm,
        ])->jsRender();
    }
``
and with grid buttons works, but i think is a big change any change that refer to callbacks and must be tested more extensively

**Comment by ibelar** [comment]
There is a fix for this issue: #653 

Would not remove confirm option for atkAjaxec since other function might rely on it.
